sdk/core/azure-core-tracing-opentelemetry/tests/test_tracing_implementations.py

```diff
@@ -43,7 +43,7 @@ def test_no_span_passed_in_with_no_environ(self, tracer):
 
     def test_span(self, tracer):
         with tracer.start_as_current_span("Root") as parent:
-            assert OpenTelemetrySpan.get_current_tracer() is tracer
+            assert OpenTelemetrySpan.get_current_tracer().__class__ is tracer.__class__
```

Comment:
That's not the good test, you're testing that the type are consistent, but I would expect the instances to be consistent too, not just the class. Why is that change?

Reply:
Changed this test a little bit. Tested to make sure the underlying instance of TracerSource which is returned is the same.